### DISCUSSION TITLE
Welcome to ch32-hal Discussions!

### DISCUSSION THREAD

**Discussion by andelf** [ch32-rs/ch32-hal#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by romainreignier** [comment]
Hi,

I am Romain, I love to tinker with microcontrollers on my spare time and I have some interest in Rust, especially in embedded context where I find it amazing that the embedded-hal has managed to standardize the drivers development with its traits.
Sadly, my time available to work on this kind of projects is limited so I have not done any real project yet using rust-embedded.

After collecting stm32 boards for its great software support within several open-source projects for some time, I have then tried some ESP32 thanks to the great rust support done by Espressif. I have now started to try the WCH CH32V line with the cheap CH32V003 but also the CH32V307 for its Ethernet peripheral (not yet investigated). The CH32X033 might be the next one for the cheap price and the USB peripheral.

Anyway, thanks a lot @andelf for this nice project that look promising but using all the modern features of embedded-hal and embassy.

I might try to help a bit here with some examples but I am not comfortable with the code in the HAL yet.

**Reply by andelf** [reply]
Feel free to add any kind of contributions.
And better add a WIP PR after a skeleton or template is done(Then I'll know what you are working with, to avoid conflicts).

Any simple demo code is also welcomed.
